### PR TITLE
E2E: Add tests for test-infra-definition code

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Add tests for test-infra-definition code for [VM options](https://pkg.go.dev/github.com/DataDog/test-infra-definitions@main/scenarios/aws/vm/ec2params) and for [the Agent options](https://pkg.go.dev/github.com/DataDog/test-infra-definitions@main/components/datadog/agentparams).

Improve the invoke task `new_e2e_tests`:
- Add code coverage support
- Add a test_run_name to select the tests to run.

Note: This PR requires https://github.com/DataDog/test-infra-definitions/pull/311 to be merged first.
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/test-infra-definition/cross_platform_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		if os == ec2os.CentOS || os == ec2os.RockyLinux {
+			// TODO: There is a bug for CentOS and RockyLinux
+			continue
+		}
+
+		s.subTestInstalAgent(os)
+	}
+}
+
+func (s *crossPlatformSuite) subTestInstalAgent(os ec2os.Type) {
```
Comment (pducolin) [inline test/new-e2e/test-infra-definition/cross_platform_test.go:58]: **🥜 nitpick**
Typo in the name
```suggestion
func (s *crossPlatformSuite) subTestInstallAgent(os ec2os.Type) {
```

File: test/new-e2e/test-infra-definition/cross_platform_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	s.T().Run(fmt.Sprintf("Test install agent %v", os), func(t *testing.T) {
+		s.UpdateEnv(e2e.AgentStackDef([]ec2params.Option{ec2params.WithOS(os)}))
+		output := s.Env().Agent.Status().Content
+		require.Contains(s.T(), output, "Agent start")
```
Comment (pducolin) [inline test/new-e2e/test-infra-definition/cross_platform_test.go:62]: **❓ question**
Why not `assert` here ?

File: test/new-e2e/test-infra-definition/cross_platform_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	})
+}
+
+func (s *crossPlatformSuite) subTestInstallAgentVersion(os ec2os.Type) {
```
Comment (pducolin) [inline test/new-e2e/test-infra-definition/cross_platform_test.go:66]: **💬 suggestion**
This test is asserting we can upgrade from version 6 to version 7, naming could be more explicit

**Review by pducolin — APPROVED** [review]
Thank you for adding these, some of them can be actual tests used to ensure we can upgrade the agent on different OSes

File: test/new-e2e/test-infra-definition/cross_platform_test.go (hunk 1 above)
Comment (ogaca-dd) [reply test/new-e2e/test-infra-definition/cross_platform_test.go]: Code updated

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file.

**Pull request merged**